core/domain/image_validation_services.py

```diff
@@ -23,13 +23,18 @@
 from core import utils
 from core.domain import html_validation_service
 
+from typing import Optional, Union
 
-def validate_image_and_filename(raw_image, filename):
+
+def validate_image_and_filename(
+    raw_image: Optional[Union[str, bytes]],
+    filename: Optional[str]
```

Comment:
union of `str` and `bytes` is added because this method is used in the controller and there we are accepting basestring ( a type for both str and bytes ). Also, None is added ( by using optional ) because we are handling the `None` case of the `raw_image` in this method's code.

the `filename` is also annotated with `optional[str]` because for `filename` we are also handling the `None` case in mthod's code.